src/braket/aws/aws_session.py

```diff
@@ -825,3 +825,27 @@ def copy_session(
         # Preserve user_agent information
         copied_session._braket_user_agents = self._braket_user_agents
         return copied_session
+
+    def get_full_image_tag(self, image_uri: str) -> str:
+        """
+        Get verbose image tag from image uri.
+
+        Args:
+            image_uri (str): Image uri to get tag for.
+
+        Returns:
+            str: Verbose image tag for given image.
+        """
+        registry = image_uri.split(".")[0]
+        repository, tag = image_uri.split("/")[-1].split(":")
+        digest = self.ecr_client.batch_get_image(
```

Comment:
Can we loop through the returned images and look for the first tag that is not `latest` and use that?  This would save us a second call to ECR